server/src/main.rs

```diff
@@ -344,114 +312,93 @@ impl Backend {
         )
     }
 
-    async fn get_sources_directory(
+    async fn get_reference_directories(
         &self,
-        workspace_root: &Option<Url>,
+        root_uri: &Option<Url>,
         config_params: Option<&DidChangeConfigurationParams>,
-    ) -> Result<Url, Option<tower_lsp::jsonrpc::Error>> {
-        let sources_directory = if let Some(params) = config_params {
-            params
-                .settings
-                .get("slice")
-                .and_then(|v| v.get("sourceDirectory"))
-                .and_then(|v| v.as_str())
-                .map(|s| s.to_string())
-                .ok_or(None)?
-        } else {
-            let params = vec![ConfigurationItem {
-                scope_uri: None,
-                section: Some("slice.sourceDirectory".to_string()),
-            }];
-
-            let result = self.client.configuration(params).await?;
-            result
-                .get(0)
-                .and_then(|v| v.as_str().map(|s| s.to_string()))
-                .ok_or(None)?
+    ) -> Result<Vec<Url>, Option<tower_lsp::jsonrpc::Error>> {
+        // If no root_uri is set, we cannot resolve the reference directories
+        let Some(root_uri) = root_uri else {
+            return Err(Some(tower_lsp::jsonrpc::Error::invalid_params(
+                "Root directory is not set.",
+            )));
         };
 
-        if let Some(workspace_dir) = workspace_root {
-            let workspace_path = workspace_dir.to_file_path().map_err(|_| {
-                Some(tower_lsp::jsonrpc::Error::invalid_params(
-                    "Failed to convert workspace URL to file path.",
-                ))
-            })?;
+        let reference_directories = if let Some(params) = config_params {
+            SliceConfig::from_configuration_parameters(params.clone(), root_uri)
+                .reference_urls
+                .unwrap_or_default()
+        } else {
+            SliceConfig::try_from_backend(self, root_uri)
+                .await?
+                .reference_urls
+                .unwrap_or_default()
```

Comment:
Are these unwrap or defaults connect? If the paths are messed up for some reason, or the we're unable to fetch something from the client, we should probably complain instead of using unexpected defaults.